gnovm/stdlibs/std/native.gno

```diff
@@ -1,14 +1,15 @@
 package std
 
-// AssertOriginCall panics if [IsOriginCall] returns false.
-func AssertOriginCall() // injected
-
-// IsOriginCall returns true only if the calling method is invoked via a direct
-// MsgCall. It returns false for all other cases, like if the calling method
+// AssertOriginCall panics if the calling method is not invoked via a direct
+// MsgCall. It doesn't panic for any other cases, like if the calling method
 // is invoked by another method (even from the same realm or package).
-// It also returns false every time when the transaction is broadcasted via
+// It also panic every time when the transaction is broadcasted via
 // MsgRun.
-func IsOriginCall() bool // injected
+func AssertOriginCall() {
+	if !PrevRealm().IsUser() {
+		panic("invalid non-origin call")
+	}
+}
```

Comment:
Actually, we want to keep AssertOriginCall in its current behaviour (I disagree with it, but alas). ie., it should panic according to the rules of IsOriginCall().

Ie., AssertOriginCall should panic if in the same realm A calls B, and B contains `AssertOriginCall()`.

`AssertOriginCall()` should also always panic for all usages within `maketx run`.

Reply:
🤯🤯

Done here : https://github.com/gnolang/gno/pull/3399/commits/d55c5a0e7508258b65385c57764034dc45fbf918